pysiaf/utils/tools2.py

```diff
@@ -0,0 +1,160 @@
+"""A collection of helper functions to support pysiaf calculations and functionalities
+
+Authors
+-------
+
+    - Colin Cox
+
+References
+----------
+
+
+
+"""
+
+import sys
+import numpy as np
+from math import *
+import copy
+import pylab as pl
+
+from astropy.table import Table
+
+from ..aperture import PRD_REQUIRED_ATTRIBUTES_ORDERED
+from .polynomial import poly, ShiftCoeffs, Rotate, FlipY, FlipX, rotate_coefficients, RotateCoeffs, triangle, invert
+from ..siaf import Siaf
+from pysiaf.aperture import DISTORTION_ATTRIBUTES
+from pysiaf.utils import tools
+
+
+def match_v2v3(aperture_1, aperture_2, verbose=False):
+    """ Use the V2V3 from aperture_1 in aperture_2  modifying XDetRef, YDetReƒf,
+    XSciRef YSciRef to match
+    Also  copy or shift the polynomial coefficients to reflect the new reference point origin
+    and for NIRCam recalculate angles. """
+
+    instrument_codes = {'NRC':'NIRCam', 'MIR':'MIRI', 'FGS':'FGS', 'NRS':'NIRSpec', 'NIS':'NIRISS'}
+    code = aperture_1.AperName[:3]
+    instrument = instrument_codes[code]
```

Comment:
you can get rid of `instrument_codes` and use
`instrument=aperture_1.InstrName` (InstrName is all upper case)